airbyte-integrations/connectors/source-dynamodb/src/main/java/io/airbyte/integrations/source/dynamodb/DynamodbUtils.java

```diff
@@ -40,8 +43,26 @@ public static DynamoDbClient createDynamoDbClient(final DynamodbConfig dynamodbC
       LOGGER.info("Creating credentials using access key and secret key");
       AwsCredentials awsCreds = AwsBasicCredentials.create(dynamodbConfig.accessKey(), dynamodbConfig.secretKey());
       awsCredentialsProvider = StaticCredentialsProvider.create(awsCreds);
+    } else if (!StringUtils.isBlank(dynamodbConfig.roleArn())) {
+      LOGGER.info("Using Role Based Access with provided role ARN");
+
+      AssumeRoleRequest assumeRoleRequest = AssumeRoleRequest.builder()
+        .roleArn(dynamodbConfig.roleArn())
+        .roleSessionName("airbyte-source-dynamodb")
+        .build();
+
+      StsClient stsClient = StsClient.builder()
+        .region(dynamodbConfig.region())
```

Comment:
This probably needs to be behind a conditional checking if region has been set, similar to line 73 in this file.